dsperse/src/compile/compiler.py

```
@@ -108,36 +263,54 @@ def _compile_slices(self, dir_path: str, input_file_path: Optional[str] = None,
                 if 'compilation' not in slice_data or not isinstance(slice_data.get('compilation'), dict):
                     slice_data['compilation'] = {}
-                slice_data['compilation']['ezkl'] = comp_block
+                slice_data['compilation'][used_backend] = comp_block
 
             # Save model-level metadata (or single slice metadata)
             Utils.save_metadata_file(metadata, os.path.dirname(metadata_path), os.path.basename(metadata_path))
 
-        logger.info(f"Compilation of slices completed. Compiled {compiled_count} slices, skipped {skipped_count} slices.")
+        # Log summary
+        backend_summary = {}
+        for idx, backend in backend_stats.items():
```

Ruff (0.14.7)

Loop control variable `idx` not used within loop body

(B007)